dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+
+            // Unwrap the launch/tests error now, if there was one
+            result?;
+        }
+    }
+
+    Ok(())
+}
+
+fn unpack_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'unpack'").run()?;
+    let tgz_path = &env.a4x2_package_tar;
+
+    if !tgz_path.try_exists()? {
+        bail!("a4x2-package bundle does not exist at {}, did you run `cargo xtask a4x2-package`?", tgz_path);
+    }
+
+    cmd!(sh, "tar -xvzf {tgz_path}").run()?;
+
+    if !env.a4x2_dir.try_exists()? {
+        bail!("extracting a4x2-package bundle did not result in a4x2-package-out/ existing");
+    }
+
+    let a4x2_dir = &env.a4x2_dir;
+    cmd!(sh, "tar -cf sled-common.tar -C {a4x2_dir}/cargo-bay/sled-common ./")
+        .run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g0").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g1").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g2").run()?;
+    cmd!(sh, "tar -xf sled-common.tar -C {a4x2_dir}/cargo-bay/g3").run()?;
+
+    Ok(())
+}
+
+fn prepare_to_launch_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'prepare'").run()?;
+
+    // TODO could move these into the a4x2-package bundle.
+    // mostly of matter of having a consistent version of propolis.
+    exec_remote_script(&sh, "https://raw.githubusercontent.com/oxidecomputer/falcon/main/get-ovmf.sh")?;
+    exec_remote_script(&sh, "https://raw.githubusercontent.com/oxidecomputer/falcon/main/get-propolis.sh")?;
+
+    // Generate an ssh key we will use to log into the sleds.
+    cmd!(sh, "ssh-keygen -t ed25519 -N '' -f a4x2-ssh-key").run()?;
+
+    // Copy the public side into the cargo bay
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g0/root_authorized_keys"),
+    )?;
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g1/root_authorized_keys"),
+    )?;
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g2/root_authorized_keys"),
+    )?;
+    sh.copy_file(
+        "a4x2-ssh-key.pub",
+        env.a4x2_dir.join("cargo-bay/g3/root_authorized_keys"),
+    )?;
+
+    Ok(())
+}
+
+fn try_launch_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    let _popdir = sh.push_dir(&env.a4x2_dir);
+
+    cmd!(sh, "pfexec ./a4x2 launch").run()?;
+
+    // XXX Is there a better way to do this?
+    let ce_addr_json =
+        cmd!(sh, "./a4x2 exec ce 'ip -4 -j addr show enp0s10'").read()?;
+
+    // Translated from this jq query:
+    // .[0].addr_info[] | select(.dynamic == true) | .local
+    let ce_addr_json: Value = serde_json::from_str(&ce_addr_json)?;
+    let customer_edge_addr = &ce_addr_json[0]["addr_info"]
+        .as_array()
+        .unwrap()
+        .iter()
+        .find(|v| v["dynamic"] == Value::Bool(true))
+        .ok_or(anyhow!("failed to find customer edge addr"))?["local"]
+        .as_str()
+        .unwrap();
```

Comment:
May as well replace these `unwrap()`s with error messages?
Would it make sense to define a Rust struct that derives `Deserialize` so that if any of the assumptions in `ce_addr_json[0]["addr_info"].as_array().unwrap()` aren't true, we fail gracefully?  (value might not be an array; might not have `addr_info` property, it might not be an array, etc.)